velox/common/memory/Memory.h

```diff
@@ -126,6 +126,23 @@ static inline MemoryPool& getCheckedReference(std::weak_ptr<MemoryPool> ptr) {
 };
 } // namespace detail
 
+class ScopedMemory {
```

Comment:
I think you can use AlignedBuffer instead.

Reply:
AlignedBuffer does free the memory automatically when reference count goes to zero.